### ISSUE TITLE
msgpack error when the event contains TABPAGE_OBJ

### ISSUE THREAD

**Issue by dzhou121** [neovim/go-client#15]
Trying to use this client with the external tabline (https://github.com/neovim/neovim/pull/6583) which will have a TABPAGE_OBJ in the message. The decoding doesn't seem to work properly.

**Comment by garyburd** [comment]
Please show the code involved, what happened and what you expected.

**Comment by dzhou121** [comment]
Thanks for your reply. Does the msgpack lib support extension types?

**Comment by justinmk** [comment]
@dzhou121 there is some test coverage in https://github.com/neovim/go-client/blob/master/nvim/ext_test.go

**Comment by garyburd** [comment]
Yes, the package supports extension types.  Please share some code and describe what's happening.

**Comment by garyburd** [comment]
TABPAGE_OBJ values are decoded to type [nvim.Tabpage](https://godoc.org/github.com/neovim/go-client/nvim#Tabpage).

I noticed in the comments at https://github.com/neovim/neovim/pull/6583 that issue goes away when INTEGER_OBJ is used instead of TABPAGE_OBJ.  The problem may be that the Go application is expecting an integer where an nvim.Tabpage is used instead.

**Comment by dzhou121** [comment]
Please see the code below, I simply print it. And the output is:

```
[tabline_update <nil>]
fatal error: all goroutines are asleep - deadlock!
```
The output if I change ```TABPAGE_OBJ``` to ```INTEGER_OBJ```:
```
[tabline_update [1 [map[tab:1 name:[No Name]]]]]
```

```
package main

import (
	"fmt"
	"os"

	"github.com/neovim/go-client/nvim"
)

func main() {
	neovim, err := nvim.NewEmbedded(&nvim.EmbedOptions{
		Args: os.Args[1:],
		Path: "./nvim",
	})
	if err != nil {
		return
	}

	done := make(chan struct{})

	neovim.RegisterHandler("redraw", func(updates ...[]interface{}) {
		for _, update := range updates {
			fmt.Println(update)
		}
	})

	go neovim.Serve()

	o := make(map[string]interface{})
	o["rgb"] = true
	o["ext_popupmenu"] = true
	o["ext_tabline"] = true
	neovim.AttachUI(100, 20, o)
	<-done
}
```

**Comment by dzhou121** [comment]
I think I've found the issue. Please review https://github.com/neovim/go-client/pull/16

**Comment by garyburd** [comment]
Fixed by #16.
